test_dataset.py

```diff
@@ -21,6 +25,13 @@ def __init__(self, config):
         self.filelist = self.create_filelist()
         self.tokenizer = Tokenizer(self.config)
 
+        self.partitions = flatten(
```

Comment:
Can we call this file_partitions? And would be good to explain the schema of the tuple we store in here, so it's easy to follow the code.

Reply:
converted into named tuple to make it clearer even though it doesn't get used as a named tuple